File: ResourceManager/Helpers/MutexLockHelper.cs
using System.Diagnostics;
using System.Threading;

namespace ResourceManager.Helpers
{
    internal static class MutexLockHelper
    {
        private static Mutex mutex = null!;

        public static bool CreateMutex()
        {
            if (mutex != null)
            {
                return true;
            }

            mutex = new Mutex(true, GetCurrentProcessName(), out bool createdNew);

            if (createdNew)
            {
                return true;
            }
            else
            {
                mutex = null!;
                return false;
            }
        }

        public static void ReleaseMutex()
        {
            if (mutex == null)
            {
                return;
            }

            mutex.ReleaseMutex();
        }

        private static string GetCurrentProcessName()
        {
            using (var currentProcess = Process.GetCurrentProcess())
            {
                return currentProcess.ProcessName;
            }
        }
    }
}

File: ResourceManager/Helpers/MutexLockHelper.vb
Imports System.Diagnostics
Imports System.Threading

Namespace ResourceManager.Helpers
    Friend Module MutexLockHelper
        Private mutex As System.Threading.Mutex = Nothing

        Public Function CreateMutex() As Boolean
            If MutexLockHelper.mutex IsNot Nothing Then
                Return True
            End If

            Dim createdNew As Boolean = Nothing
            MutexLockHelper.mutex = New Mutex(True, GetCurrentProcessName(), createdNew)

            If createdNew Then
                Return True
            Else
                ResourceManager.Helpers.MutexLockHelper.mutex = Nothing
                Return False
            End If
        End Function

        Public Sub ReleaseMutex()
            If MutexLockHelper.mutex Is Nothing Then
                Return
            End If

            Call MutexLockHelper.mutex.ReleaseMutex()
        End Sub

        Private Function GetCurrentProcessName() As String
            Using currentProcess = Process.GetCurrentProcess()
                Return currentProcess.ProcessName
            End Using
        End Function
    End Module
End Namespace
